IBPSA/Electrical/BaseClasses/PV/BaseClasses/PartialPVElectricalSingleDiode.mo

```diff
@@ -0,0 +1,98 @@
+within IBPSA.Electrical.BaseClasses.PV.BaseClasses;
+partial model PartialPVElectricalSingleDiode
+  "Partial electrical model for PV module model following the single diode approach"
+  extends IBPSA.Electrical.BaseClasses.PV.BaseClasses.PartialPVElectrical(
+      redeclare IBPSA.Electrical.Data.PV.SingleDiodeData dat);
+
+  replaceable parameter Data.PV.SingleDiodeData dat
+    constrainedby IBPSA.Electrical.Data.PV.SingleDiodeData
+    "PV Panel data definition" annotation (choicesAllMatching);
+
+// Parameters from module data sheet
+
+protected
+  final parameter Modelica.Units.SI.Efficiency eta0=dat.eta0
+    "Efficiency under standard conditions";
+
+  final parameter Modelica.Units.SI.Area APan=dat.APan
+    "Area of one Panel, must not be confused with area of the whole module";
+
+  final parameter Modelica.Units.SI.Voltage VOC0=dat.VOC0
+    "Open circuit voltage under standard conditions";
+
+  final parameter Modelica.Units.SI.ElectricCurrent ISC0=dat.ISC0
+    "Short circuit current under standard conditions";
+
+  final parameter Modelica.Units.SI.Voltage VMP0=dat.VMP0
+    "MPP voltage under standard conditions";
+
+  final parameter Modelica.Units.SI.ElectricCurrent IMP0=dat.IMP0
+    "MPP current under standard conditions";
+
+  final parameter Modelica.Units.SI.Power PMax=dat.PMP0*1.05
+    "Maximal power of one PV module under standard conditions. P_MPP with 5 perc tolerance. This is used to limit DCOutputPower.";
+
+  final parameter Real TCoeISC(unit="A/K") = dat.TCoeISC
+    "Temperature coefficient for short circuit current, >0";
+
+  final parameter Real TCoeVOC(unit="V/K") = dat.TCoeVOC
+    "Temperature coefficient for open circuit voltage, <0";
+
+  final parameter Modelica.Units.SI.LinearTemperatureCoefficient alphaISC=
+  dat.alphaISC "Normalized temperature coefficient for short circuit current, >0";
+
+  final parameter Modelica.Units.SI.LinearTemperatureCoefficient betaVOC=
+  dat.betaVOC "Normalized temperature coefficient for open circuit voltage, <0";
+
+  final parameter Modelica.Units.SI.LinearTemperatureCoefficient gammaPMP=
+  dat.gammaPMP "Normalized temperature coefficient for power at MPP";
```

Comment:
Please format as done on lines 23 to 40.

Reply:
done